internal/agent/event_sender.go

```diff
@@ -320,14 +324,20 @@ func (sender *metricsIngestSender) sendBatches() {
 		select {
 
 		case batch := <-sender.batchQueue:
+			ctx := goContext.Background()
+			ctx, txn := instrumentation.SelfInstrumentation.StartTransaction(ctx, "sender.sendBatches")
```

Comment:
What do you think to have one constants file with all transaction names listed there? Would be easier to modify namespace/subsystem/metric names

Reply:
Sounds good 👍

Reply:
I'm just thinking that we might want to have some dynamic (samples.sample, plugins.plugin ...) maybe we should wait until we have a list of what we want to report and then decide?

Reply:
+1 with the idea of having a file for transactions/spans names. The dynamic ones could be generated of a constant prefix + dynamic name.

The file could also be used to know what are we instrumenting.

Reply:
I like the idea, it's just that I don't want to go too far this as this is just a POC. I'll create a doc to track this ideas/doubts and when it get's prioritised we can discuss these. wdyt?